pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -218,6 +224,22 @@ func GetPipelineConditionStatus(prName string, state []*ResolvedPipelineRunTask,
 		}
 	}
 	if !allFinished {
+		if !startTime.IsZero() && pipelineTimeout != nil {
+			timeout := pipelineTimeout.Duration
+			runtime := time.Since(startTime.Time)
+			if runtime > timeout {
```

Comment:
this logic to determine if a timeout has occurred seems to me like a good candidate for moving into a separate function with some tests (e.g. `hasTimedOut`)